src/H5FDs3comms.c

```diff
@@ -1091,6 +1096,12 @@ H5FD_s3comms_s3r_open(const char *url, const char *region, const char *id, const
         if (handle->signing_key == NULL)
             HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, NULL, "could not malloc space for handle key copy.");
         H5MM_memcpy(handle->signing_key, signing_key, tmplen);
+
+        tmplen        = HDstrlen(token) + 1;
+        handle->token = (char *)H5MM_malloc(sizeof(char) * tmplen);
```

Comment:
Needs a corresponding `H5MM_xfree` in `H5FD_s3comms_s3r_close`

Reply:
This has been added.

Reply:
This has been added. Also found another place where it is be added.